model/qti/CustomInteractionAsset/Extractor/TextReaderAssetExtractor.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\qti\CustomInteractionAsset\Extractor;
+
+use oat\taoQtiItem\model\qti\CustomInteractionAsset\Extractor\Api\AssetExtractorInterface;
+use oat\taoQtiItem\model\qti\interaction\CustomInteraction;
+
+/**
+ * @author Kiryl Poyu <kyril.poyu@taotesting.com>
+ */
+class TextReaderAssetExtractor implements AssetExtractorInterface
+{
+    public const INTERACTION_IDENTIFIER = 'textReaderInteraction';
```

Comment:
can you add more details to your comment?

Reply:
When you put a public constant, anyone can use it in another class or part of the code, that creates dependency on your code and if you change that constant it can break your code.

If you need to use it in your test, you only need to copy the value in the test, never put the reference to the constant. If someone changes the code, the test should break.

If you need the value to use it in various parts of the code, you could create an Enum (for now only available in version 8.1) or create a ValueObject that would behave like an Enum.

Reply:
I understand a concept. But i don't think it always necessary increase complexity of component in this way. it's fair for `CONTENT_PREFIX` which created in general for internal usage.